api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -148,18 +149,29 @@ public int compareTo(CohortMembership o) {
 	/**
 	 * @since 2.3.0
 	 * Indicates if a given cohortMembership object is equal to this one
-	 * 
-	 * @param otherCohortMembership is a CohortMembership object that should be checked for equality with this object
+	 *
+	 * @param otherCohortMembershipObject is a CohortMembership object that should be checked for equality with this object
 	 * @return true if both objects are logically equal. This is the case when endDate, startDate and patientId are equal  
 	 */
-	public boolean equals(CohortMembership otherCohortMembership) {
+	@Override
+	public boolean equals(Object otherCohortMembershipObject) {
+		if(!(otherCohortMembershipObject instanceof CohortMembership)){
```

Comment:
Wouldn't it be cheaper for the `otherCohortMembership != null` check to be checked early here than at the `return` statement?

Reply:
@samuelmale  Thank you for the review. I add the != null check to the top of the function! 

Do you have any further remarks